lib/sproxyd.js

```diff
@@ -370,10 +370,17 @@ class SproxydClient {
     batchDelete(list, reqUids, callback) {
         assert.strictEqual(typeof list, 'object');
         assert(list.keys.every(k => k.length === 40));
-        const log = this.createLogger(reqUids);
-        const payload = Buffer.from(JSON.stringify(list));
-        this._failover('POST', null, payload.length, '.batch_delete', 0, log,
-            callback, {}, payload);
+        // split the list into batches of 1000 each
+        const batches = [];
+        while (list.keys.length > 0) {
+            batches.push({ keys: list.keys.splice(0, 999) });
+        }
+        async.mapLimit(batches, 5, (b, done) => {
```

Comment:
since you don't use the result of the async functions, you can use `eachLimit` instead.